packages/flutter/lib/src/painting/alignment.dart

```diff
@@ -0,0 +1,611 @@
+  ///
+  /// If either is null, this function interpolates from [Alignment.center], and
+  /// the result is an object of the same type as the non-null argument.
+  ///
+  /// If [lerp] is applied to two objects of the same type ([Alignment] or
+  /// [AlignmentDirectional]), an object of that type will be returned (though
+  /// this is not reflected in the type system). Otherwise, an object
+  /// representing a combination of both is returned. That object can be turned
+  /// into a concrete [Alignment] using [resolve].
+  static AlignmentGeometry lerp(AlignmentGeometry a, AlignmentGeometry b, double t) {
+    if (a == null && b == null)
+      return null;
+    if (a == null)
+      return b * t;
+    if (b == null)
+      return a * (1.0 - t);
+    if (a is Alignment && b is Alignment)
+      return Alignment.lerp(a, b, t);
+    if (a is AlignmentDirectional && b is AlignmentDirectional)
+      return AlignmentDirectional.lerp(a, b, t);
+    return new _MixedAlignment(
+      ui.lerpDouble(a._x, b._x, t),
+      ui.lerpDouble(a._start, b._start, t),
+      ui.lerpDouble(a._y, b._y, t),
+    );
+  }
+
+  /// Convert this instance into a [Alignment], which uses literal
+  /// coordinates (the `x` coordinate being explicitly a distance from the
+  /// left).
+  ///
+  /// See also:
+  ///
+  ///  * [Alignment], for which this is a no-op (returns itself).
+  ///  * [AlignmentDirectional], which flips the horizontal direction
+  ///    based on the `direction` argument.
+  Alignment resolve(TextDirection direction);
+
+  @override
+  String toString() {
+    if (_start == 0.0)
+      return Alignment._stringify(_x, _y);
+    if (_x == 0.0)
+      return AlignmentDirectional._stringify(_start, _y);
+    return Alignment._stringify(_x, _y) + ' + ' + AlignmentDirectional._stringify(_start, 0.0);
+  }
+
+  @override
+  bool operator ==(dynamic other) {
+    if (other is! AlignmentGeometry)
+      return false;
+    final AlignmentGeometry typedOther = other;
+    return _x == typedOther._x &&
+           _start == typedOther._start &&
+           _y == typedOther._y;
+  }
+
+  @override
+  int get hashCode => hashValues(_x, _start, _y);
+}
+
+/// An offset that's expressed as a fraction of a [Size].
+///
+/// `Alignment(1.0, -1.0)` represents the top right of the [Size].
+///
+/// `Alignment(-1.0, 1.0)` represents the bottom left of the [Size].
+///
+/// `Alignment(0.0, 3.0)` represents a point half way across the [Size],
+/// below the bottom of the rectangle by the height of the [Size].
+///
+/// The [Alignment] class specifies offsets in terms of a distance from
+/// the top left, regardless of the [TextDirection]. To support both
+/// left-to-right and right-to-left layouts, consider using
+/// [AlignmentDirectional], which is expressed in terms of an offset from
+/// the leading edge, which is then resolved in terms of a [TextDirection]
+/// (typically obtained from the ambient [Directionality]).
```

Comment:
this paragraph is a big lie as far as I can tell :-)

Reply:
Done.